ts/features/zendesk/screens/ZendeskSeeReportsRouters.tsx

```diff
@@ -0,0 +1,190 @@
+        params: { assistanceForPayment }
+      });
+    } else {
+      navigation.navigate("ZENDESK_MAIN", {
+        screen: "ZENDESK_CHOOSE_CATEGORY",
+        params: { assistanceForPayment }
+      });
+    }
+  };
+
+  const continueButtonProps = {
+    testID: "continueButtonId",
+    bordered: false,
+    onPress: handleContactSupportPress,
+    title: I18n.t("support.helpCenter.cta.contactSupport")
+  };
+
+  const cancelButtonProps = {
+    testID: "cancelButtonId",
+    bordered: true,
+    onPress: () => navigation.goBack(),
+    title: I18n.t("global.buttons.back")
+  };
+
+  return (
+    <SafeAreaView style={IOStyles.flex} testID={"emptyTicketsComponent"}>
+      <InfoScreenComponent
+        image={renderInfoRasterImage(image)}
+        title={I18n.t("support.ticketList.noTicket.title")}
+        body={I18n.t("support.ticketList.noTicket.body")}
+      />
+      <FooterStackButton buttons={[continueButtonProps, cancelButtonProps]} />
+    </SafeAreaView>
+  );
+};
+
+type Props = IOStackNavigationRouteProps<
+  ZendeskParamsList,
+  "ZENDESK_ASK_SEE_REPORTS_PERMISSIONS"
+>;
+/**
+ * this screen checks if a user has at least a ticket, it shows:
+ * - a loading state when the request start
+ * - the list of the ticket if the user has at least a ticket in the history
+ * - an empty request screen if the user has not ticket
+ * @constructor
+ */
+const ZendeskSeeReportsRouters = (props: Props) => {
+  const dispatch = useDispatch();
+  const zendeskToken = useIOSelector(zendeskTokenSelector);
+  const ticketNumber = useIOSelector(zendeskTicketNumberSelector);
+  const assistanceForPayment = props.route.params.assistanceForPayment;
+
+  const [zendeskConfig, setZendeskConfig] = React.useState<ZendeskAppConfig>(
+    zendeskToken
+      ? { ...zendeskDefaultJwtConfig, token: zendeskToken }
+      : zendeskDefaultAnonymousConfig
+  );
+
+  useEffect(() => {
+    setZendeskConfig(
+      zendeskToken
+        ? { ...zendeskDefaultJwtConfig, token: zendeskToken }
+        : zendeskDefaultAnonymousConfig
+    );
+  }, [zendeskToken]);
+
+  useEffect(() => {
+    initSupportAssistance(zendeskConfig);
+
+    // In Zendesk we have two configuration: JwtConfig and AnonymousConfig.
+    // The AnonymousConfig is used for the anonymous user.
+    // Since the zendesk session token and the profile are provided by two different endpoint
+    // we sequentially check both:
+    // - if the zendeskToken is present the user will be authenticated via jwt
+    // - nothing is available (the user is not authenticated in IO) the user will be totally anonymous also in Zendesk
+    const zendeskIdentity = pipe(
+      zendeskToken,
+      O.fromNullable,
+      O.map((zT: string): JwtIdentity | AnonymousIdentity => ({
+        token: zT
+      })),
+      O.getOrElse(() => ({}))
+    );
+
+    setUserIdentity(zendeskIdentity);
+    dispatch(zendeskRequestTicketNumber.request());
+  }, [dispatch, zendeskConfig, zendeskToken]);
```

Comment:
In this case, we can remove the local state `zendeskConfig` and the first `useEffect`.
So we can manage everything for Zendesk in the second `useEffect`.
For example:

```typescript
useEffect(() => {
    const zendeskConfig = pipe(
      zendeskToken,
      O.fromNullable,
      O.map(
        (zT: string): ZendeskAppConfig => ({
          ...zendeskDefaultJwtConfig,
          token: zT
        })
      ),
      O.getOrElseW(() => zendeskDefaultAnonymousConfig)
    );

    initSupportAssistance(zendeskConfig);

    // In Zendesk we have two configuration: JwtConfig and AnonymousConfig.
    // The AnonymousConfig is used for the anonymous user.
    // Since the zendesk session token and the profile are provided by two different endpoint
    // we sequentially check both:
    // - if the zendeskToken is present the user will be authenticated via jwt
    // - nothing is available (the user is not authenticated in IO) the user will be totally anonymous also in Zendesk
    const zendeskIdentity = pipe(
      zendeskToken,
      O.fromNullable,
      O.map((zT: string): JwtIdentity | AnonymousIdentity => ({
        token: zT
      })),
      O.getOrElseW(() => ({}))
    );

    setUserIdentity(zendeskIdentity);
  }, [dispatch, zendeskToken]);
```
what do you think?